examples/ipv6only-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,639 @@
+        return pw::Status::InvalidArgument();
+    case ESP_ERR_ESP_NETIF_INVALID_PARAMS:
+        return pw::Status::InvalidArgument();
+    case ESP_ERR_WIFI_IF:
+        return pw::Status::NotFound();
+    case ESP_ERR_WIFI_NOT_CONNECT:
+        return pw::Status::FailedPrecondition();
+    case ESP_ERR_WIFI_NOT_STARTED:
+        return pw::Status::FailedPrecondition();
+    case ESP_ERR_WIFI_CONN:
+        return pw::Status::Internal();
+    case ESP_FAIL:
+        return pw::Status::Internal();
+    default:
+        return pw::Status::Unknown();
+    }
+}
+
+} // namespace pw::internal
+
+namespace {
+
+constexpr pw::containers::FlatMap<uint8_t, uint32_t, 73> kChannelToFreqMap({ {
+    { 1, 2412 },   { 2, 2417 },   { 3, 2422 },   { 4, 2427 },   { 5, 2432 },   { 6, 2437 },   { 7, 2442 },   { 8, 2447 },
+    { 9, 2452 },   { 10, 2457 },  { 11, 2462 },  { 12, 2467 },  { 13, 2472 },  { 14, 2484 },  { 32, 5160 },  { 34, 5170 },
+    { 36, 5180 },  { 38, 5190 },  { 40, 5200 },  { 42, 5210 },  { 44, 5220 },  { 46, 5230 },  { 48, 5240 },  { 50, 5250 },
+    { 52, 5260 },  { 54, 5270 },  { 56, 5280 },  { 58, 5290 },  { 60, 5300 },  { 62, 5310 },  { 64, 5320 },  { 68, 5340 },
+    { 96, 5480 },  { 100, 5500 }, { 102, 5510 }, { 104, 5520 }, { 106, 5530 }, { 108, 5540 }, { 110, 5550 }, { 112, 5560 },
+    { 114, 5570 }, { 116, 5580 }, { 118, 5590 }, { 120, 5600 }, { 122, 5610 }, { 124, 5620 }, { 126, 5630 }, { 128, 5640 },
+    { 132, 5660 }, { 134, 5670 }, { 136, 5680 }, { 138, 5690 }, { 140, 5700 }, { 142, 5710 }, { 144, 5720 }, { 149, 5745 },
+    { 151, 5755 }, { 153, 5765 }, { 155, 5775 }, { 157, 5785 }, { 159, 5795 }, { 161, 5805 }, { 165, 5825 }, { 169, 5845 },
+    { 173, 5865 }, { 183, 4915 }, { 184, 4920 }, { 185, 4925 }, { 187, 4935 }, { 188, 4940 }, { 189, 4945 }, { 192, 4960 },
+    { 196, 4980 },
+} });
+
+constexpr size_t kUdpBufferSize = 512;
+
+constexpr uint8_t kWiFiConnectRetryMax = 5;
+
+/* The event group allows multiple bits for each event, but we only care about
+ * whether the STA initialization was successful
+ */
+constexpr uint8_t WIFI_STA_UP_BIT = BIT0;
+
+/* The event group allows multiple bits for each event, but we only care about 3 events:
+ * - we are connected to the AP with an IP6
+ * - we are connected to the AP with an IP4
+ * - we failed to connect after the maximum amount of retries */
+#define WIFI_CONNECTED_IP4_BIT BIT1
+#define WIFI_CONNECTED_IP6_BIT BIT2
+#define WIFI_FAIL_BIT BIT3
+
+SemaphoreHandle_t wifi_init_semaphore;
+SemaphoreHandle_t scanresults_semaphore;
+
+/* FreeRTOS event group to signal when we are connected*/
+EventGroupHandle_t s_wifi_event_group;
+
+int s_retry_num        = 0;
+esp_netif_t * wifi_sta = nullptr;
+
+int sockfd = 0;
+unsigned int clientlen;         // byte size of client's address
+struct sockaddr_in6 clientaddr; // client addr
+
+::pw::rpc::ServerWriter<chip_rpc_ScanResults> scanresult_writer;
+
+class RAIIGuard
+{
+    std::function<void()> release_;
```

Comment:
nit: Can you move this to a new `private:` section after the `public:` section please?

Reply:
done